### PR TITLE
Xonotic: gLibc incompatibility

### PR DESCRIPTION
**Pull request by markmandel**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking

/kind bug

> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

Xonotic image was previously broken with the following error:

```
❯ kubectl logs xonotic-ggbpd-g22ww --container xonotic
/home/xonotic/wrapper: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /home/xonotic/wrapper)
/home/xonotic/wrapper: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /home/xonotic/wrapper)
```

Looks like it needed a newer version of debian to run inline with the updated golang update.

**Which issue(s) this PR fixes**:

Should let #3494 pass once the new image is pushed to prod

**Special notes for your reviewer**:

Glad we're building the e2e tests now!

### CODE REVIEW

**Comment by markmandel** [comment]
@Kalaiselvi84 can you please push this to production once it's approved and merged. It should unblock you on #3494

**Review by zmerlynn — APPROVED** [review]

**Comment by markmandel** [comment]
Here is a test run of the image locally to prove it works:

<details>

<summary>docker run...</summary>

```
❯ docker run --network=host xonotic-example:1.4-linux-amd64
>>> Connecting to Agones with the SDK
>>> Starting health checking
>>> Starting wrapper for Xonotic!
>>> Path to Xonotic server script: /home/xonotic/Xonotic/server_linux.sh []
Game is Xonotic using base gamedir data
gamename for server filtering: Xonotic
Xonotic Linux 03:14:05 Jun 20 2023 - release
Current nice level is below the soft limit - cannot use niceness
Skeletal animation uses SSE code path
execing quake.rc
execing default.cfg
execing xonotic-common.cfg
execing xonotic-client.cfg
Warning: Could not expand $vid_width
Warning: Could not expand $vid_height
Warning: Could not expand $vid_pixelheight
Warning: Could not expand $vid_desktopfullscreen
execing sRGB-disable.cfg
execing binds-xonotic.cfg
execing effects-normal.cfg
execing crosshairs.cfg
execing gamemodes-client.cfg
execing notifications.cfg
execing _hud_common.cfg
execing _hud_descriptions.cfg
execing hud_luma.cfg
execing xonotic-server.cfg
execing testing.cfg
execing balance-xonotic.cfg
execing bal-wep-xonotic.cfg
execing physicsX.cfg
execing turrets.cfg
execing vehicles.cfg
execing gamemodes-server.cfg
execing ctfscoring-samual.cfg
execing mutators.cfg
execing randomitems-xonotic.cfg
execing monsters.cfg
execing minigames.cfg
execing physics.cfg
execing cdtracks.cfg
execing commands.cfg
Warning: Could not expand $qport in alias _if_dedicated
execing help.cfg
couldn't exec config.cfg
"maxplayers" set to "16"
execing data/campaign.cfg
execing config_update.cfg
execing font-xolonium.cfg
Warning: Could not expand $scr_loadingscreen_barheight
Warning: Could not expand $scr_infobar_height
This command only works on clients: cl_cmd hud scoreboard_columns_set
execing autoexec/empty.cfg
execing autoexec.cfg
Loaded public key key_0.d0pk (fingerprint: Xon//KssdlzGkFKdnnN4sgg8H+koTbBn5JTi37BAW1Q=)
Generated private ID key_0.d0pk (public key fingerprint: Qt6Rxh+rkbU3TZP7A70Pdbd4AX+Oi/Hof9Q7+l/CYFo=)
Saved unsigned key to key_0.d0si
curl failed
execing post-config.cfg
>>> Found 'listening' statement: 2
Server using port 26000
Server listening on address 0.0.0.0:26000
Server listening on address [0:0:0:0:0:0:0:0]:26000
execing server.cfg
maxplayers can not be changed while a server is running.
It will be changed on next server startup ("map" command).
"maxplayers" set to "8"
Game type successfully switched to dm
Maplist contains no usable maps!  Resetting it to default map list.
Switching to map xoylent
menu: unknown program name
>>> Found 'listening' statement: 4
>>> Moving to READY: Server using port 26000
Server listening on address 0.0.0.0:26000
Server listening on address [0:0:0:0:0:0:0:0]:26000
Server using port 26000
Server listening on address 0.0.0.0:26000
Server listening on address [0:0:0:0:0:0:0:0]:26000
```

</details>

**Comment by Kalaiselvi84** [comment]
> @Kalaiselvi84 can you please push this to production once it's approved and merged. It should unblock you on #3494

yes, will do it👍🏻

**Review by gongmax — APPROVED** [review]

**Pull request merged**
